File: app/src/main/java/shop/ineed/app/ineed/adapter/SlideAdapter.java
package shop.ineed.app.ineed.adapter;

import android.content.Context;
import android.os.Parcelable;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.squareup.picasso.Picasso;

import java.util.List;

import shop.ineed.app.ineed.R;

/**
 * Created by jose on 9/9/17.
 *
 * Class gerenciadora, slide.
 */

public class SlideAdapter extends PagerAdapter {

    private List<String> mImages;
    private LayoutInflater mInflater;
    private Context mContext;


    public SlideAdapter(Context context, List<String> images) {
        this.mContext = context;
        this.mImages = images;
        this.mInflater = LayoutInflater.from(mContext);
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((View) object);
    }

    @Override
    public int getCount() {
        return mImages.size();
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        View view = mInflater.inflate(R.layout.slide_adapter, container, false);

        ImageView imageSlide = (ImageView) view.findViewById(R.id.imageSlide);

        Picasso.with(mContext).load(mImages.get(position)).into(imageSlide);

        container.addView(view, 0);

        return view;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view.equals(object);
    }

    @Override
    public Parcelable saveState() {
        return null;
    }
}

File: app/src/main/java/shop/ineed/app/ineed/adapter/SlideAdapter.kt
package shop.ineed.app.ineed.adapter

import android.content.Context
import android.os.Parcelable
import android.support.v4.view.PagerAdapter
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.slide_adapter.view.*

import com.squareup.picasso.Picasso

import shop.ineed.app.ineed.R
import shop.ineed.app.ineed.interfaces.RecyclerClickListener

/**
 * Created by jose on 9/9/17.
 *
 * Class gerenciadora, slide.
 */

class SlideAdapter(private val mContext: Context, private val mImages: List<String>, var onClick: RecyclerClickListener) : PagerAdapter() {

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }

    override fun getCount(): Int {
        return mImages.size
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = LayoutInflater.from(mContext).inflate(R.layout.slide_adapter, container, false)

        with(view){
            Picasso.with(mContext).load(mImages[position]).into(imageSlideViewPhotoAdapter)
            Log.d("SLIDE", mImages[position])

            setOnClickListener{ onClick.onClickRecyclerListener(view, position) }
            container.addView(view, 0)
        }


        return view
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view == `object`
    }

    override fun saveState(): Parcelable? {
        return null
    }

    fun setOnClickListener(onClick: RecyclerClickListener){
        this.onClick = onClick
    }
}
